File: backend/Models/SharedTrip.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    public enum PermissionLevel
    {
        View,
        Edit
    }

    public class SharedTrip
    {
        [Key]
        public long SharedTripId { get; set; }
        public long TripId { get; set; }

        public Trip? Trip { get; set; }

        public long UserId { get; set; }

        public User? User { get; set; }

        [Required]
        public PermissionLevel PermissionLevel { get; set; }
    }
} 

File: frontend/src/models/shared-trip.ts
export const PermissionLevel = {
	View: "View",
	Edit: "Edit",
} as const;

export type PermissionLevel =
	(typeof PermissionLevel)[keyof typeof PermissionLevel];

export interface SharedTrip {
	sharedTripId: number;
	tripId: number;
	userId: number;
	permissionLevel: PermissionLevel;
}
